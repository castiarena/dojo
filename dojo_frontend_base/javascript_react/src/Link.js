import React from 'react';

const status = {
    NORMAL: 'link-normal',
    HOVERED: 'link-hovered',
};

export default class Link extends React.Component{
    constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            className: status.NORMAL
        };
    }

    _onMouseEnter() {
        this.setState({className : status.HOVERED})
    }

    _onMouseLeave() {
        this.setState({className : status.NORMAL})
    }

    render() {
        return (
            <a
            className={this.state.className}
            href={this.props.page}
            onMouseEnter={this._onMouseEnter}
            onMouseLeave={this._onMouseLeave}>
                {this.props.children}
            </a>
        );
    }
}