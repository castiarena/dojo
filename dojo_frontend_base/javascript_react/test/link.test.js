import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import * as enzyme from 'enzyme';

import Link from '../src/Link';

enzyme.configure({ adapter: new Adapter() });

describe('Link tests', () => {
    it('should be in normal status', () => {
        const component = renderer.create(
            <Link page="google.com.ar">Google argentina</Link>
        ).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should be in hovered status', () => {
        const component = shallow(
            <Link page="google.com.ar">Google argentina</Link>
        );

        component.find('a').simulate('mouseEnter');

        expect(component.find('a').hasClass('link-hovered')).toBeTruthy();
    });

    it('should be in normal status after mouse enter and leave', () => {
        const component = shallow(
            <Link page="google.com.ar">Google argentina</Link>
        );

        component.find('a').simulate('mouseEnter');
        component.find('a').simulate('mouseLeave');
        expect(component.find('a').hasClass('link-normal')).toBeTruthy();
    });
});