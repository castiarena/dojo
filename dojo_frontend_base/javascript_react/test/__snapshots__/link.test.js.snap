// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link tests should be in normal status 1`] = `
<a
  className="link-normal"
  href="google.com.ar"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google argentina
</a>
`;
